docs/application/dotnet/guides/machine-learning/pipeline.md

```diff
@@ -0,0 +1,180 @@
+  Using the various nodes and functionalities in Pipeline, you can easily handle and control the data flow.
+
+## Prerequisites
+
+To enable your application to use the Machine Learning Inference API functionality:
+
+1. To use the methods and properties of the `Tizen.MachineLearning.Inference.Pipeline` class or its related classes such as `Tizen.MachineLearning.Inference.TensorsData` and `Tizen.MachineLearning.Inference.TensorsInfo`, include the `Tizen.MachineLearning.Inference` namespace in your application:
+
+    ```C#
+    using Tizen.MachineLearning.Inference;
+    ```
+
+2. If the model file you want to use is located in the **media storage** or the **external storage**, the application has to request permission by adding the following privileges to the `tizen-manifest.xml` file:
+
+    ```xml
+    <privileges>
+      <!--To access media storage-->
+      <privilege>http://tizen.org/privilege/mediastorage</privilege>
+
+      <!--To access, read, and write to the external storage-->
+      <privilege>http://tizen.org/privilege/externalstorage</privilege>
+    </privileges>
+    ```
+
+<a name="construct"></a>
+## Pipeline construction
+
+To construct a pipeline, you must have a pipeline description with the [GStreamer](https://gstreamer.freedesktop.org) and [NNStreamer](https://github.com/nnstreamer/nnstreamer) elements.
+For more information, see the GStreamer manual or the NNStreamer documentation for examples and grammar.
+
+1. If the model file is located in the resource directory of your own application, you need to get its absolute path:
+
+    ```C#
+    string ResourcePath = Tizen.Applications.Application.Current.DirectoryInfo.Resource;
+    string model_path = ResourcePath + "models/mobilenet_v1_1.0_224_quant.tflite";
+    ```
+
+2. You can construct a pipeline with the description string, including the neural network framework and a model file:
+
+    ```C#
+    /* Create Pipeline instance with the pipeline description. */
+    string description = "appsrc name=srcx ! " +
+                         "other/tensor,dimension=(string)3:224:224:1,type=(string)uint8,framerate=(fraction)0/1 ! " +
+                         "tensor_filter framework=tensorflow-lite model=" + model_path + " ! " +
+                         "tensor_sink name=sinkx";
+    Pipeline pipe = new Pipeline(description);
+    ```
+
+<a name="inputdata"></a>
+## Pushing the input data
+
+`SourceNode` provides a method to push input data into the pipeline:
+
+```C#
+/* Get the source node with the name 'srcx'. */
+var src_node = pipe.GetSource("srcx");
+
+/* Start the pipeline. */
+pipe.Start();
+
+/* Prepare input tensor data, with data type UInt8 and dimension 3:224:224:1. */
+var in_info = new TensorsInfo();
+in_info.AddTensorInfo(TensorType.UInt8, new int[4] { 3, 224, 224, 1 });
+
+var in_data = in_info.GetTensorsData();
+
+/* Push input tensor data into the pipeline. */
+src_node.Input(in_data);
+```
+
+<a name="outputdata"></a>
+## Subscribing to the output data
```

Comment:
Thank you for updating the headings in sentence case.

Global Comment:
It is recommended to use "sentence case" and "verb format" for headings. No need to add extra articles.
Please do not use the gerund form (ing form).
For example,
## Subscribe to output data

Reply:
Sorry, my mistake. I just updated this PR. Please see the changes again.

Reply:
Thank you for updating the changes.